src/main/java/com/manoelcampos/javadoc/coverage/configuration/Configuration.java

```diff
@@ -0,0 +1,124 @@
+            Optional.of("javadoc-coverage"));
+
+    private static final List<Option<?>> ALL_OPTIONS = new ArrayList<>();
+
+    /**
+     * Static constructor for initializing the options list
+     */
+    static {
+        ALL_OPTIONS.add(computePublicCoverageOnly);
+        ALL_OPTIONS.add(coverageFileName);
+    }
+
+    /**
+     * The raw options which need to be parsed to our configuration
+     */
+    private final String[][] rawOptions;
+
+    public Configuration(String[][] rawOptions) {
+        this.rawOptions = rawOptions;
+    }
+
+    public boolean computePublicCoverageOnly() {
+        return isOptionContained(computePublicCoverageOnly);
+    }
+
+    public String getCoverageFileName() {
+        if (isOptionContained(coverageFileName)) {
+            return getOptionValue(coverageFileName);
+        }
+        return coverageFileName.getDefaultValue();
+    }
+
+    /**
+     * Gets the output directory passed as a command line argument to javadoc tool.
+     *
+     * @return the output directory to export the JavaDocs
+     */
+    public String getOutputDirectory() {
+        // This is no option of the doclet, but instead of the javadoc tool
+        // so we don't declare it as an option here directly)
+
+        for (final String[] option : rawOptions) {
+            if (option.length == 2 && option[0].equals("-d")) {
+                return Utils.includeTrailingDirSeparator(option[1]);
+            }
+        }
+
+        return "";
+    }
+
+    private boolean isOptionContained(Option<?> option) {
+        for (final String[] opt : rawOptions) {
+            if (option.isOption(opt[0])) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private <T> T getOptionValue(Option<T> option) {
+        if (!isOptionContained(option) && !option.hasDefaultValue()) {
+            throw new IllegalStateException("Option is not contained and has no default value");
+        }
+
+        if (!isOptionContained(option)) {
+            return option.getDefaultValue();
+        }
+
+        for (final String[] opt: rawOptions) {
+            if (option.isOption(opt[0])) {
+                return option.parseValue(opt);
+            }
+        }
+
+        throw new IllegalStateException("no valid option found although it should be there");
+    }
+
+    /**
+     * This method is necessary for validating the doclet parameters, which happens before instantiating the
+     * CoverageDoclet and the configuration container.
+     *
+     * @param option
+     * @return the length of the option
+     */
+    public static int getOptionLength(final String option) {
+        Optional<Option<?>> opt = ALL_OPTIONS.stream().filter(o -> o.isOption(option)).findFirst();
+        if (opt.isPresent()) {
+            return opt.get().getLength();
+        }
+        return Standard.optionLength(option);
+    }
```

Comment:
Using `isPresent` is a bad practice also. You could replace the lines starting at the `if` by:

`return opt.map(Option::getLength).orElseGet(() -> Standard.optionLength(option))`